community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -472,6 +479,115 @@ def update_topology(lkp: util.Lookup) -> None:
         log.debug("Topology configuration updated. Reconfiguring Slurm.")
         util.scontrol_reconfigure(lkp)
 
+
+def delete_reservations(reservation_map: Dict[str, str]) -> None:
+    for reservation_name in list(reservation_map.keys()):
+      util.run(SLURM_DELETE_RESERVATION.format(reservation_name), timeout=30)
+      del reservation_map[reservation_name]
+
+
+def update_slurm_reservation_maintenance(reservation_name: str, node: str,
+                                         starttime: str, reservation_map: Dict[str, str]) -> None:
+    ## If reservation exists for maintenance, update reservation if time window
+    ## for scheduled maintenance changes.
+    if reservation_map is not None and reservation_name in reservation_map:
+      details = reservation_map[reservation_name]
+
+      nodes = None
+      start_time = None
+
+      values = details.split(",")
+      nodes = values[0].strip()
+      maintenance_start_time = values[1].strip()
+
+      if nodes and start_time:
+        if maintenance_start_time == starttime:
+          return
+
+        util.run(SLURM_DELETE_RESERVATION.format(reservation_name), timeout=30)
+        util.run(SLURM_CREATE_RESERVATION.format(starttime, nodes, reservation_name), timeout=30)
+
+      del reservation_map[reservation_name]  # remove the reservation from the map.
+
+    else: # Reservation doesn't exist, make reservation during maintenance window.
+      util.run(SLURM_CREATE_RESERVATION.format(starttime, node, reservation_name), timeout=30)
+
+
+def get_slurm_reservation_maintenance() -> Dict[str, str]:
+    res = util.run(SLURM_SHOW_RESERVATION, timeout=30)
+    all_reservations = [x.split() for x in res.stdout.split("\n\n")[:-1]]
+    reservation_map = {}
+
+    for reservation in all_reservations:
+        reservation_name = None
+        nodes = None
+        start_time = None
+        for item in reservation:
+          key, value = item.split('=', 1)  # Split at the first '='
+
+          if key == 'ReservationName':
+            reservation_name = value
+          elif key == 'Nodes':
+            nodes = value
+          elif key == 'StartTime':
+            start_time = value
+
+        if reservation_name is None or nodes is None or start_time is None:
+          continue
+
+        # Check if the reservation is for scheduled maintenance.
+        maintenance_reservation = nodes + "_maintenance"
+        if reservation_name != maintenance_reservation:
+          continue
+
+        reservation_map[reservation_name] = nodes + ', ' + start_time
```

Comment:
Why do you compose a string here instead of returning `tuple[str,whatever_type_for_start_time]` ?